contracts/System.template

```diff
@@ -110,6 +111,13 @@ contract System {
 
   modifier openForAll() {_;}
 
+  modifier nonReentrant() {
```

Comment:
I recommend being more defensive when non-secure contracts (which for us is effectively every non-platform contract) are interacted with and simply use guards.

I am aware of the fact that these functions are arranged in checks-effects-interactions mode but such an arrangement may break on new changes, a smart hacker may try cross-reentry and, and in general, why not play safe?

For reasons of storage compatibility we will need to use a bool flag and not a uint at least if we want to make it into a common code.
This will make the guard operation more expensive - 50K gas vs. 30K - but I believe it's worth the added security. I will also create a tech-debt here and switch to uint guard once the additional base-storage problem is resolved.